### ISSUE TITLE
Application to Extreme Multi-label Classification?

### ISSUE THREAD

**Issue by onclue** [Microsoft/EdgeML#74]
Hi, 

Thank you for this great work. I am curious about how to apply EdgeML to extreme multi-label classification problems. Your documentation for ProtoNN Wiki and paper suggests that it is a good fit but it is not clear how to get started with that. In the C++ code, what should the data format be for multi-label problems? In the tensorflow versions, how should the model be updated to apply it to multi-label problems (how to define the loss and output layer, etc..). Also, is only ProtoNN suitable for XML problems, or can the other algorithms be adapted too?

Cheers!

**Comment by adityakusupati** [comment]
Hi,

Thanks for the interest. @AIgen  can comment on the C++ ProtoNN for Extreme Classification. 
Coming to tensorflow, I think it is as simple as changing the final loss from multi-class to a multi-label loss and train it normally as done right now. You can use all the 3 (ProtoNN, Bonsai and FastGRNN) for extreme classification, but only ProtoNN has been well tested. FastGRNN runs on raw-text and has a 1-vs-All classifier at the end.

Please refer to the extreme classification repository for the datasets, It should be straight forward to change the codes accordingly. But, as the datasets are sparse, tf ProtoNN code might not be efficient on GPU.

Let us know if there is any other issue.
